internal/metrics/collector/settings.go

```diff
@@ -62,15 +68,47 @@ func createConverterFactories() []confmap.ConverterFactory {
 }
 
 func createURIs(cfg *config.Config) []string {
-	return []string{getConfig(cfg)}
+	return []string{cfg.Metrics.CollectorConfigPath}
 }
 
-func getConfig(_ *config.Config) string {
-	val, ex := os.LookupEnv("OPENTELEMETRY_COLLECTOR_CONFIG_FILE")
-	if !ex {
-		return "/tmp/otel-collector-config.yaml"
+// Generates a OTel Collector config to a file by injecting the Metrics Config to a Go template.
+func writeCollectorConfig(conf *config.Metrics) error {
+	otelcolTemplate, err := template.New(otelTemplatePath).ParseFiles(otelTemplatePath)
+	if err != nil {
+		return err
+	}
+
+	// Should check that we are within an allowed directory in the future.
+	confPath := filepath.Clean(conf.CollectorConfigPath)
+
+	// Check if file exists.
+	_, err = os.Stat(confPath)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return err
+		}
+
+		// Create if doesn't exist.
+		_, createErr := os.Create(confPath)
+		if createErr != nil {
+			return createErr
+		}
+	}
+
+	file, err := os.OpenFile(confPath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, configFilePermission)
+	if err != nil {
+		return err
+	}
+
+	err = otelcolTemplate.Execute(file, conf)
+	if err != nil {
+		return err
+	}
+
+	err = file.Close()
```

Comment:
should we use `defer` when closing the file? In the case where `otelcolTemplate.Execute(file, conf)` fails we won't close the file

Reply:
Good point! The reason why I didn't use `defer` here is that the static security code scan tool we use will complain about not checking `file.Close()` errors. I guess we can check it in the `defer` but it then cannot be returned.

Perhaps if I log the error in the event that it happens in the `defer`, then that will work as a middle ground?

Reply:
Added the following `defer`:
```go
defer func() {
  err = file.Close()
  if err != nil {
    slog.Warn("Failed to close file", "file_path", confPath)
  }
}()
```